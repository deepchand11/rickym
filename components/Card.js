import React, { memo } from "react";
import { StyleSheet, Text, View, Image } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";
import { Octicons } from "@expo/vector-icons";
import { statusIndicator } from "../common/utils";

const Card = ({ item, navigation }) => {
  return (
    <View style={styles.card}>
      <Image
        style={styles.tinyLogo}
        source={{
          uri: item.image,
        }}
      />
      <View style={styles.content}>
        <Text numberOfLines={1} style={styles.title}>
          {item.name}
        </Text>
        <View style={{ flexDirection: "row" }}>
          <Octicons
            name="primitive-dot"
            size={14}
            color={statusIndicator(item.status)}
          />
          <Text numberOfLines={1} style={styles.status}>
            {item.status} - {item.species}
          </Text>
        </View>
        <Text style={styles.subtitle}>Last known location:</Text>
        <TouchableOpacity
          onPress={() =>
            navigation.navigate("Details", {
              url: item.location.url,
            })
          }
          testID="locationId"
        >
          <Text numberOfLines={1} style={styles.subtext}>
            {item.location.name}
          </Text>
        </TouchableOpacity>
        <Text style={styles.subtitle}>First seen in:</Text>
        <TouchableOpacity
          onPress={() =>
            navigation.navigate("Episode", {
              url: item.episode[0],
            })
          }
          testID="episodeId"
        >
          <Text numberOfLines={1} style={styles.subtext}>
            episode
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default memo(Card);

const styles = StyleSheet.create({
  card: {
    flexDirection: "row",
    backgroundColor: "rgb(60, 62, 68)",
    marginVertical: 8,
    marginHorizontal: 16,
    borderRadius: 20,
    flex: 1,
  },
  tinyLogo: {
    width: 115,
    height: 115,
    borderTopLeftRadius: 20,
    borderBottomLeftRadius: 20,
  },
  content: {
    padding: 5,
    paddingLeft: 10,
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
    color: "#ffffff",
    width: 180,
  },
  status: {
    fontSize: 11,
    color: "rgb(158, 158, 158)",
    marginLeft: 3,
  },
  subtitle: {
    fontSize: 11,
    color: "rgb(158, 158, 158)",
  },
  subtext: {
    fontSize: 12,
    color: "#ffffff",
    textDecorationLine: "underline",
    width: 180,
  },
});
