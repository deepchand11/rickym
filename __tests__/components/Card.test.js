import "react-native";
import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react-native";
import Card from "../../components/Card";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("navigates on location press", async () => {
  const navigate = jest.fn();
  const item = {
    image: "imageUrl",
    location: {
      name: "locationName",
      url: "locationUrl",
    },
    episode: ["episodeUrl"],
  };
  const { getByTestId } = await render(
    <Card item={item} navigation={{ navigate }} />
  );
  const locationId = getByTestId("locationId");
  fireEvent.press(locationId);
  expect(navigate).toHaveBeenCalledWith("Details", { url: "locationUrl" });
});
it("navigates on episode press", async () => {
  const navigate = jest.fn();
  const item = {
    image: "imageUrl",
    location: {
      name: "locationName",
      url: "locationUrl",
    },
    episode: ["episodeUrl"],
  };
  const { getByTestId } = await render(
    <Card item={item} navigation={{ navigate }} />
  );
  const locationId = getByTestId("episodeId");
  fireEvent.press(locationId);
  expect(navigate).toHaveBeenCalledWith("Episode", { url: "episodeUrl" });
});
it("Card renders correctly", () => {
  const navigate = jest.fn();
  const item = {
    image: "imageUrl",
    location: {
      name: "locationName",
      url: "locationUrl",
    },
    episode: ["episodeUrl"],
  };
  const tree = renderer
    .create(<Card item={item} navigation={{ navigate }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
