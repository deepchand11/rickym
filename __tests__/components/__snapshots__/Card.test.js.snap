// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(60, 62, 68)",
      "borderRadius": 20,
      "flex": 1,
      "flexDirection": "row",
      "marginHorizontal": 16,
      "marginVertical": 8,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "imageUrl",
      }
    }
    style={
      Object {
        "borderBottomLeftRadius": 20,
        "borderTopLeftRadius": 20,
        "height": 115,
        "width": 115,
      }
    }
  />
  <View
    style={
      Object {
        "padding": 5,
        "paddingLeft": 10,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 20,
          "fontWeight": "bold",
          "width": 180,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#E4E4E4",
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontFamily": "octicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï†‚
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "rgb(158, 158, 158)",
            "fontSize": 11,
            "marginLeft": 3,
          }
        }
      >
         - 
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "rgb(158, 158, 158)",
          "fontSize": 11,
        }
      }
    >
      Last known location:
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      testID="locationId"
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 12,
              "textDecorationLine": "underline",
              "width": 180,
            }
          }
        >
          locationName
        </Text>
      </View>
    </RNGestureHandlerButton>
    <Text
      style={
        Object {
          "color": "rgb(158, 158, 158)",
          "fontSize": 11,
        }
      }
    >
      First seen in:
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      testID="episodeId"
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 12,
              "textDecorationLine": "underline",
              "width": 180,
            }
          }
        >
          episode
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
